scripts/plot.py

```diff
@@ -0,0 +1,151 @@
+    for i in range(0, len(backends)):
+        for j in range(0, len(num_sources[i])):
+            index = unified_num_sources.index(num_sources[i][j])
+            data[i, index] = (num_sources[i][j] / timings[i][j])/10**6
+            errors_data[i, 0, index] = num_sources[i][j]/10**6*(1/timings[i][j] - 1/(timings[i][j]+errors[i][j]))
+            errors_data[i, 1, index] = num_sources[i][j]/10**6*(1/(timings[i][j]-errors[i][j]) - 1/timings[i][j])
+
+    # Phase 2: plot generation
+    ax = plt.subplot(111)
+    patterns = ('/', 'x', '\\', '.', 'o', 'O')
+
+    def autolabel(rects):
+        """Attach a text label above each bar in *rects*, displaying its height."""
+        for rect in rects:
+            height = round(rect.get_height(),1)
+            ax.annotate('{}'.format(height),
+                        xy=(rect.get_x() + rect.get_width() / 2, height-rect.get_height()/2),
+                        xytext=(0, 3),  # 3 points vertical offset
+                        textcoords="offset points",
+                        ha='center', va='bottom')
+
+    width = 0.6
+    scale = 1/len(backends)
+    for i in range(0, len(backends)):
+        barplot = ax.bar(np.arange(len(unified_num_sources)) + scale*(i-.5*(len(backends)-1)) * width,
+                         width=scale*width*np.ones(len(unified_num_sources)),
+                         height=data[i, :],
+                         yerr=errors_data[i, :],
+                         capsize=5,
+                         hatch = patterns[i % len(patterns)],
+                         edgecolor='k',
+                         label=backends[i])
+        if numbers:
+            autolabel(barplot)
+
+    ax.legend(loc='upper left', fontsize=20)
+
+    ax.set_xlabel('Number of objects indexed', fontsize=20)
+    plt.xticks(fontsize=18)
+    ax.set_xticks(np.arange(len(num_source_labels)))
+    ax.set_xticklabels(num_source_labels)
+    ax.tick_params(axis='x', which='major', bottom='off', top='off')
+    ax.set_xlim([-0.5, len(unified_num_sources)-0.5])
+
+    ax.set_ylabel('Rate (million objects/second)', fontsize=20)
+    plt.yticks(fontsize=18)
+    ax.yaxis.grid('on')
+
+if __name__ == '__main__':
+
+    # Process input
+    options = docopt(__doc__)
+
+    algorithm = options['--algo']
+    backends = options['--backends']
+    geometry = options['--geometry']
+    input_files = options['--input-files']
+    numbers = options['--numbers']
+    output_file = options['--output-file']
+
+    plt.figure(figsize=(6, 5))
+
+    # Check input
+    if geometry != 'filled_box' and geometry != 'hollow_box':
+        raise Exception('Unknown geometry: "' + geometry + '"')
+    if algorithm != 'construction' and algorithm != 'knn_search' and algorithm != 'radius_search':
+        raise Exception('Unknown algorithm: "' + algorithm + '"')
+
+    backends = backends.split(',')
+    input_files = input_files[0]
+    input_files = input_files.split(',')
```

Comment:
I think that if you use docopt's `INPUT...`, you can do `-i file1 -i file2 -i files3` automatically producing a list, so you don't need this.

Reply:
I can try that. Specifying input files via `-i file1,file2,file3` was my first intuition so I tried to make that work. I don't care much one way or the other, though.

Reply:
Yes, commenting these two lines `-i file1 -i file2 -i files3` works. Do you prefer that syntax?

Reply:
Yes, this way we rely on docopt and no write the code ourselves. Could do the same with backends.

Reply:
Why would you design interfaces rather for ease of implementation than for ease of use?
Anyway, I don't feel strongly it so I'll just change.

Reply:
Here you go!